packages/core/src/transport/httpRequest.ts

```diff
@@ -11,11 +13,15 @@ import { addFailedSendBeacon } from './failedSendBeacon'
  * to be parsed correctly without content type header
  */
 export class HttpRequest {
-  constructor(private endpointBuilder: EndpointBuilder, private bytesLimit: number) {}
+  sendWithRetry: (data: string | FormData, bytesCount: number) => Observable<Response>
+
+  constructor(private endpointBuilder: EndpointBuilder, private configuration: Configuration) {
+    ;({ retried: this.sendWithRetry } = httpRetry(this.send, configuration))
```

Comment:
**❗ warning:** ‏that does not seem right:

- httpRetry depends on a HttpRequest method 
- httpRequest depends on httpRetry